diff --git a/src/stream/3rdparty/stb_vorbis.c b/src/stream/3rdparty/stb_vorbis.c
index 1181e6d..a548942 100644
--- a/src/stream/3rdparty/stb_vorbis.c
+++ b/src/stream/3rdparty/stb_vorbis.c
@@ -65,6 +65,7 @@
 #ifndef STB_VORBIS_NO_STDIO
 #include <stdio.h>
 #endif
+#include <base/types.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -881,7 +882,7 @@ static int error(vorb *f, enum STBVorbisError e)
 #ifdef dealloca
 #define temp_free(f,p)                  (f->alloc.alloc_buffer ? 0 : dealloca(size))
 #else
-#define temp_free(f,p)                  0
+#define temp_free(f,p)
 #endif
 #define temp_alloc_save(f)              ((f)->temp_offset)
 #define temp_alloc_restore(f,p)         ((f)->temp_offset = (p))
@@ -3047,7 +3048,7 @@ typedef int16 YTYPE;
 #else
 typedef int YTYPE;
 #endif
-static int do_floor(vorb *f, Mapping *map, int i, int n, float *target, YTYPE *finalY, uint8 *step2_flag)
+static int do_floor(vorb *f, Mapping *map, int i, int n, float *target, YTYPE *finalY, VOID(uint8 *step2_flag))
 {
    int n2 = n >> 1;
    int s = map->chan[i].mux, floor;
@@ -3154,7 +3155,7 @@ static int vorbis_decode_initial(vorb *f, int *p_left_start, int *p_left_end, in
    return TRUE;
 }
 
-static int vorbis_decode_packet_rest(vorb *f, int *len, Mode *m, int left_start, int left_end, int right_start, int right_end, int *p_left)
+static int vorbis_decode_packet_rest(vorb *f, int *len, Mode *m, int left_start, VOID(int left_end), int right_start, int right_end, int *p_left)
 {
    Mapping *map;
    int i,j,k,n,n2;
