--- /tmp/stb_vorbis.c	2020-05-11 10:29:27.021494081 +0200
+++ stb_vorbis.c	2020-05-11 10:31:10.236120785 +0200
@@ -76,6 +76,7 @@
 #ifndef STB_VORBIS_NO_STDIO
 #include <stdio.h>
 #endif
+#include <base/types.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -567,9 +568,17 @@
 #include <stdio.h>
 #endif
 
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #ifndef STB_VORBIS_NO_CRT
-   #include <stdlib.h>
-   #include <string.h>
+   #ifdef HAVE_RMALLOC_H
+      #include <rmalloc.h>
+   #else
+      #include <stdlib.h>
+      #include <string.h>
+   #endif
    #include <assert.h>
    #include <math.h>
 
@@ -578,7 +587,9 @@
       #include <malloc.h>
    #endif
    #if defined(__linux__) || defined(__linux) || defined(__EMSCRIPTEN__)
-      #include <alloca.h>
+      #ifndef HAVE_RMALLOC_H
+         #include <alloca.h>
+      #endif
    #endif
 #else // STB_VORBIS_NO_CRT
    #define NULL 0
@@ -3050,7 +3061,7 @@
 #else
 typedef int YTYPE;
 #endif
-static int do_floor(vorb *f, Mapping *map, int i, int n, float *target, YTYPE *finalY, uint8 *step2_flag)
+static int do_floor(vorb *f, Mapping *map, int i, int n, float *target, YTYPE *finalY, UNUSED(uint8 *step2_flag))
 {
    int n2 = n >> 1;
    int s = map->chan[i].mux, floor;
@@ -3157,7 +3168,7 @@
    return TRUE;
 }
 
-static int vorbis_decode_packet_rest(vorb *f, int *len, Mode *m, int left_start, int left_end, int right_start, int right_end, int *p_left)
+static int vorbis_decode_packet_rest(vorb *f, int *len, Mode *m, int left_start, UNUSED(int left_end), int right_start, int right_end, int *p_left)
 {
    Mapping *map;
    int i,j,k,n,n2;
