--- /tmp/stb_vorbis.c	2017-08-26 07:52:42.941563053 +0200
+++ stb_vorbis.c	2017-08-26 07:58:17.244736559 +0200
@@ -66,6 +66,7 @@
 #ifndef STB_VORBIS_NO_STDIO
 #include <stdio.h>
 #endif
+#include <base/types.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -545,9 +546,17 @@
 #include <stdio.h>
 #endif
 
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #ifndef STB_VORBIS_NO_CRT
-   #include <stdlib.h>
-   #include <string.h>
+   #ifdef HAVE_RMALLOC_H
+      #include <rmalloc.h>
+   #else
+      #include <stdlib.h>
+      #include <string.h>
+   #endif
    #include <assert.h>
    #include <math.h>
 
@@ -556,7 +565,9 @@
       #include <malloc.h>
    #endif
    #if defined(__linux__) || defined(__linux) || defined(__EMSCRIPTEN__)
-      #include <alloca.h>
+      #ifndef HAVE_RMALLOC_H
+         #include <alloca.h>
+      #endif
    #endif
 #else // STB_VORBIS_NO_CRT
    #define NULL 0
@@ -883,7 +894,7 @@
 #ifdef dealloca
 #define temp_free(f,p)                  (f->alloc.alloc_buffer ? 0 : dealloca(size))
 #else
-#define temp_free(f,p)                  0
+#define temp_free(f,p)
 #endif
 #define temp_alloc_save(f)              ((f)->temp_offset)
 #define temp_alloc_restore(f,p)         ((f)->temp_offset = (p))
@@ -3049,7 +3060,7 @@
 #else
 typedef int YTYPE;
 #endif
-static int do_floor(vorb *f, Mapping *map, int i, int n, float *target, YTYPE *finalY, uint8 *step2_flag)
+static int do_floor(vorb *f, Mapping *map, int i, int n, float *target, YTYPE *finalY, VOID(uint8 *step2_flag))
 {
    int n2 = n >> 1;
    int s = map->chan[i].mux, floor;
@@ -3156,7 +3167,7 @@
    return TRUE;
 }
 
-static int vorbis_decode_packet_rest(vorb *f, int *len, Mode *m, int left_start, int left_end, int right_start, int right_end, int *p_left)
+static int vorbis_decode_packet_rest(vorb *f, int *len, Mode *m, int left_start, VOID(int left_end), int right_start, int right_end, int *p_left)
 {
    Mapping *map;
    int i,j,k,n,n2;
